bootstrap4/.js/src/main/scala/io/udash/bootstrap/dropdown/UdashDropdown.scala

```diff
@@ -22,7 +22,7 @@ final class UdashDropdown[ItemType, ElemType <: ReadableProperty[ItemType]] priv
   buttonToggle: ReadableProperty[Boolean],
   override val componentId: ComponentId
 )(
-  itemFactory: ItemType => Element,
+  itemFactory: (ItemType, Binding.NestedInterceptor) => Element,
```

Comment:
This version does not allow dynamic changes of dropdown's elements without rendering the whole menu again. What about extracting the line 81 to constructor as `Binding`? This way you could provide both factory types (`(ElemType, Binding.NestedInterceptor) => Element` and `ItemType => Element`).

Reply:
I was considering it, but that would leave a lot more responsibility to the user, especially in terms of applying the selection listener. With current approach you're free to change the content of a `DefaultDropdownItem.Dynamic` and it shouldn't trigger a full menu render (as the function reference wouldn't change). The only remaining issue would be rerendering on structure change: this can only be solved via `repeat`, which would require nesting an element again. Any suggestions?

Reply:
You could take `Binding` as an argument in this private constructor and expose two `apply` versions in the companion object with these two factory types. Inside you can hide implementation with `produce` and `repeat`.